core/domain/learner_group_services.py

```diff
@@ -613,3 +674,55 @@ def can_user_be_invited(
         )
 
     return (True, '')
+
+
+def sync_syllabus_to_stories_and_skills(
```

Comment:
Hmm, this seems like a very db call heavy function to be called whenever the syllabus is called. Can't we handle deleted story/topic elsewhere instead of doing this everytime the page is loaded?

Reply:
Have removed this sync and have added reference removal when a story or a subtopic page is deleted.